datajoint/base_relation.py

```diff
@@ -70,8 +71,14 @@ def declare(self):
         Loads the table heading. If the table is not declared, use self.definition to declare
         """
         if not self.is_declared:
-            self.connection.query(
-                declare(self.full_table_name, self.definition, self._context))
+            if self.definition is not None and self.definition is not Ellipsis:
+                self.connection.query(
+                    declare(self.full_table_name, self.definition, self._context))
+            else:
+                name = self.__class__.__name__
+                warn("""Table for %s is not declared and definition is not defined.
```

Comment:
Right, I understand that, but that's why just catch the specific error in the schema. I think it still makes sense to prevent actual usage of the class bound to undeclared table.